internal/service/autoscaling/policy.go

```diff
@@ -87,6 +87,276 @@ func ResourcePolicy() *schema.Resource {
+										Elem: &schema.Resource{
+											Schema: map[string]*schema.Schema{
+												"metric_data_queries": {
+													Type:     schema.TypeList,
+													Required: true,
+													MaxItems: 10,
+													Elem: &schema.Resource{
+														Schema: map[string]*schema.Schema{
+															"id": {
+																Type:         schema.TypeString,
+																Required:     true,
+																ValidateFunc: validation.StringLenBetween(1, 255),
+															},
+															"expression": {
+																Type:         schema.TypeString,
+																Optional:     true,
+																ValidateFunc: validation.StringLenBetween(1, 1023),
+															},
+															"metric_stat": {
+																Type:     schema.TypeList,
+																Optional: true,
+																MaxItems: 1,
+																Elem: &schema.Resource{
+																	Schema: map[string]*schema.Schema{
+																		"metric": {
+																			Type:     schema.TypeList,
+																			Required: true,
+																			MaxItems: 1,
+																			Elem: &schema.Resource{
+																				Schema: map[string]*schema.Schema{
+																					"metric_name": {
+																						Type:     schema.TypeString,
+																						Required: true,
+																					},
+																					"namespace": {
+																						Type:     schema.TypeString,
+																						Required: true,
+																					},
+																					"dimensions": {
+																						Type:     schema.TypeSet,
+																						Optional: true,
+																						Elem: &schema.Resource{
+																							Schema: map[string]*schema.Schema{
+																								"name": {
+																									Type:     schema.TypeString,
+																									Required: true,
+																								},
+																								"value": {
+																									Type:     schema.TypeString,
+																									Required: true,
+																								},
+																							},
+																						},
+																					},
+																				},
+																			},
+																		},
+																		"stat": {
+																			Type:         schema.TypeString,
+																			Required:     true,
+																			ValidateFunc: validation.StringLenBetween(1, 100),
+																		},
+																		"unit": {
+																			Type:     schema.TypeString,
+																			Optional: true,
+																		},
+																	},
+																},
+															},
+															"label": {
+																Type:         schema.TypeString,
+																Optional:     true,
+																ValidateFunc: validation.StringLenBetween(1, 2047),
+															},
+															"return_data": {
+																Type:     schema.TypeBool,
+																Optional: true,
+																Default:  true,
```

Comment:
I'm not sure if we should set a default value here because from the documentation at https://docs.aws.amazon.com/autoscaling/ec2/APIReference/API_MetricDataQuery.html, it mentions `If you are only retrieving metrics and not performing any math expressions, do not specify anything for ReturnData. This sets it to its default (true).` In the same documentation, it also specifies when it is appropriate to explicitly specify a `false`. I think we can leave this `Default: true` out. What do you think?

Reply:
Yeah, true is default for AWS API. If we remove it from here, it doesn't impact the functionality. I am open to remove it.

Reply:
I ran into this issues after removing this default value. it seems terraform will assign a `false` as default for boolean (I didn't find any doc for this behaviour) which conflict with our expected behaviour. I am will hold the default value for now and open to further discussion.